pkg/iac-providers/kustomize/commons/load-dir.go

```diff
@@ -0,0 +1,163 @@
+	"github.com/accurics/terrascan/pkg/iac-providers/output"
+	"github.com/accurics/terrascan/pkg/results"
+	"github.com/accurics/terrascan/pkg/utils"
+	"github.com/hashicorp/go-multierror"
+	"go.uber.org/zap"
+	"sigs.k8s.io/kustomize/api/filesys"
+	"sigs.k8s.io/kustomize/api/krusty"
+)
+
+const (
+	kustomizedirectory string = "kustomization"
+)
+
+var (
+	kustomizeErrMessage = "error from kustomization. error : %v"
+)
+
+// NewTerraformDirectoryLoader creates a new terraformDirectoryLoader
+func NewKustomizeDirectoryLoader(absRootDir string, options map[string]interface{}, useKustomizeBinary bool, version string) KustomizeDirectoryLoader {
+	kustomizeDirectoryLoader := KustomizeDirectoryLoader{
+		absRootDir:         absRootDir,
+		options:            options,
+		useKustomizeBinary: useKustomizeBinary,
+		version:            version,
+	}
+
+	return kustomizeDirectoryLoader
+}
+
+// LoadIacDir loads the kustomize directory and returns the ResourceConfig mapping which is evaluated by the policy engine
+func (t KustomizeDirectoryLoader) LoadIacDir() (output.AllResourceConfigs, error) {
+
+	allResourcesConfig := make(map[string][]output.ResourceConfig)
+
+	files, err := utils.FindFilesBySuffixInDir(t.absRootDir, KustomizeFileNames())
+	if err != nil {
+		zap.S().Debug("error while searching for iac files", zap.String("root dir", t.absRootDir), zap.Error(err))
+		return allResourcesConfig, multierror.Append(t.errIacLoadDirs, results.DirScanErr{IacType: "kustomize", Directory: t.absRootDir, ErrMessage: err.Error()})
+	}
+
+	if len(files) == 0 {
+		errMsg := fmt.Sprintf("kustomization.y(a)ml file not found in the directory %s", t.absRootDir)
+		zap.S().Debug("error while searching for iac files", zap.String("root dir", t.absRootDir), zap.Error(err))
+		return allResourcesConfig, multierror.Append(t.errIacLoadDirs, results.DirScanErr{IacType: "kustomize", Directory: t.absRootDir, ErrMessage: errMsg})
+	}
+
+	if len(files) > 1 {
+		errMsg := fmt.Sprintf("multiple kustomization.y(a)ml found in the directory %s", t.absRootDir)
+		zap.S().Debug("error while searching for iac files", zap.String("root dir", t.absRootDir), zap.Error(err))
+		return allResourcesConfig, multierror.Append(t.errIacLoadDirs, results.DirScanErr{IacType: "kustomize", Directory: t.absRootDir, ErrMessage: errMsg})
+	}
+
+	kustomizeFileName := *files[0]
+	yamlkustomizeobj, err := utils.ReadYamlFile(filepath.Join(t.absRootDir, kustomizeFileName))
+
+	if err != nil {
+		err = fmt.Errorf("unable to read the kustomization file in the directory %s, error: %v", t.absRootDir, err)
+		zap.S().Error("error while reading the file", kustomizeFileName, zap.Error(err))
+		return allResourcesConfig, multierror.Append(t.errIacLoadDirs, results.DirScanErr{IacType: "kustomize", Directory: t.absRootDir, ErrMessage: err.Error()})
+	}
+
+	// ResourceConfig representing the kustomization.y(a)ml file
+	config := output.ResourceConfig{
+		Name:   filepath.Dir(t.absRootDir),
```

Comment:
Shouldn't these use relative paths?

Reply:
This implementation was there from before and I have kept it as it was.